File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    visitedNumbers := make(map[int]int)

    for i, num := range(nums) {
        diff := target - num
        if idx, exists := visitedNumbers[diff]; exists {
            return []int{idx, i}
        }
        visitedNumbers[num] = i
    }

    return []int{}
}

File: 1-two-sum/two-sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut visited_numbers: HashMap<i32, i32> = HashMap::new();

        for (i, &num) in nums.iter().enumerate() {
            let diff = target - num;
            
            if let Some(&idx) = visited_numbers.get(&diff) {
                return vec![idx, i as i32];
            }

            visited_numbers.insert(num, i as i32);
        }

        vec![]
    }
}